bundles/org.openhab.binding.haywardomnilogic/src/main/java/org/openhab/binding/haywardomnilogic/internal/handler/HaywardBridgeHandler.java

```diff
@@ -0,0 +1,453 @@
+        return Collections.singleton(HaywardDiscoveryService.class);
+    }
+
+    public HaywardBridgeHandler(Bridge bridge, HttpClient httpClient) {
+        super(bridge);
+        this.httpClient = httpClient;
+    }
+
+    /**
+     * Called by the zone discovery service to let this handler have a reference.
+     */
+    public void setHaywardDiscoveryService(HaywardDiscoveryService s) {
+        this.haywardDiscoveryService = s;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void dispose() {
+        clearPolling(initializeFuture);
+        clearPolling(pollTelemetryFuture);
+        clearPolling(pollAlarmsFuture);
+        logger.trace("Hayward polling cancelled");
+        super.dispose();
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        initializeFuture = scheduler.schedule(this::scheduledInitialize, 1, TimeUnit.SECONDS);
+        return;
+    }
+
+    public void scheduledInitialize() {
+        config = getConfigAs(HaywardConfig.class);
+
+        try {
+            clearPolling(pollTelemetryFuture);
+            clearPolling(pollAlarmsFuture);
+
+            if (!(login())) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Unable to Login to Hayward's server");
+                clearPolling(pollTelemetryFuture);
+                clearPolling(pollAlarmsFuture);
+                commFailureCount = 50;
+                initPolling(60);
+                return;
+            }
+
+            if (!(getSiteList())) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Unable to getMSP from Hayward's server");
+                clearPolling(pollTelemetryFuture);
+                clearPolling(pollAlarmsFuture);
+                commFailureCount = 50;
+                initPolling(60);
+                return;
+            }
+
+            updateStatus(ThingStatus.ONLINE);
+            logger.debug("Succesfully opened connection to Hayward's server: {} Username:{}", config.endpointUrl,
+                    config.username);
+
+            initPolling(0);
+            logger.trace("Hayward Telemetry polling scheduled");
+
+            if (config.alarmPollTime > 0) {
+                initAlarmPolling(1);
+                logger.trace("Hayward Alarm polling scheduled");
+            } else {
+                logger.trace("Hayward Alarm polling disabled");
+            }
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                    "scheduledInitialize exception");
+            logger.debug("Hayward Connection thing: Unable to open connection to Hayward Server: {} Username: {}",
+                    config.endpointUrl, config.username, e);
```

Comment:
The log message could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.